src/AsyncWriter.js

```diff
@@ -390,22 +217,12 @@ AsyncWriter.prototype = {
         }
 
         lastArray.push(callback);
+        return this;
     },
 
     emit: function(type, arg) {
         var events = this._events;
-        switch(arguments.length) {
-            case 1:
-                events.emit(type);
-                break;
-            case 2:
-                events.emit(type, arg);
-                break;
-            default:
-                events.emit.apply(events, arguments);
-                break;
-        }
-
+        events.emit.apply(events, arguments);
```

Comment:
Yes, this was done for performance reasons. Might be a micro-optimization, but benchmarks showed that using a switch statement improved performance for the most common case of either 1 (just `type`) or 2 args (`type`+`arg`). Please restore the switch statement. Thanks!